third_party/intel/lib/Dialect/Triton/Transforms/RemoveMasks.cpp

```diff
@@ -444,7 +444,7 @@ class LoopVersioner {
            "Expecting a non-empty collection of masked operations");
 
     // Collect the (loop invariant) mask conditions.
-    std::set<Operation *> maskConds;
+    SmallPtrSet<Operation *, 8> maskConds;
```

Comment:
[nit] Assume usually the same mask condition if they are for boundary checks.
```suggestion
    SmallPtrSet<Operation *, 1> maskConds;
```

Reply:
A SmallPtrSet is implemented using an array if it is "small enough", so the elements are going to be added in insertion order. However when the set becomes "large enough" I am not sure whether the insertion order is preserved. So this problem might appear again (may be unlikely). 

I am ok with this change, a safer fix would be to use a container that guarantees ordering like a small vector.

Reply:
sure